reference/best_practices/reusable_bundles.rst

```diff
@@ -1,4 +1,173 @@
+
+
+In Akeneo PIM, we use `PhpSpec`_ as unit testing tool and `Behat`_ for behavior-oriented tests.
+
+Use a Continuous Integration (CI) tool to deploy your application, test your code and improve its quality.
+On our open-source projects, we use free CI tools `Travis`_ for unit tests and `Scrutinizer`_ for static analysis.
+You can find some examples on our `Akeneo-Labs`_ projects (`scrutinizer.yml`_ and `travis.yml`_).
+
+.. _Single Responsibility: https://en.wikipedia.org/wiki/Single_responsibility_principle
+.. _SOLID principles: https://en.wikipedia.org/wiki/SOLID_(object-oriented_design)
+.. _PMD: https://phpmd.org/
+.. _PhpSpec: http://phpspec.readthedocs.org/
+.. _Behat: http://docs.behat.org/
+.. _Travis: https://travis-ci.org/
+.. _Scrutinizer: https://scrutinizer-ci.com/
+.. _Akeneo-Labs: https://github.com/akeneo-labs
+.. _scrutinizer.yml: https://github.com/akeneo-labs/CustomEntityBundle/blob/master/.scrutinizer.yml
+.. _travis.yml: https://github.com/akeneo-labs/CustomEntityBundle/blob/master/.travis.yml
+
+
+Another important point about maintainability is to try to rely on interfaces not on concrete classes.
+An interface is a contract you will inject in a method. It will ease you the migration on future Akeneo PIM versions.
+
+Don't forget you can use composition instead of inheritance.
+Decorate a class will allow you to benefit of inheritance avoiding migration problems.
+Indeed, class constructors can changed from a version to another. On the other side, an interface do not change.
+
+To give to integrator a clean way to rely on your code, you can alse create interfaces.
+
+
+Extensibility
+-------------
+
+Respect Akeneo PIM dependencies
+"""""""""""""""""""""""""""""""
+You **MUST** follow Akeneo PIM dependencies at tall cost. Don't try to change version of existing ones.
+You may add new dependencies needed for your customisations.
+
+
+Use registries
+""""""""""""""
+Registries are also a good extension point when you have to manage with many cases (different attribute types, entities, etc.).
+It consists in creating a registry class and then tag services which will be add to this registry.
+
+.. note::
+    Learn more about `registry`_ pattern.
+
+You can also take example on the localizers' cookbook: :doc:`/cookbook/localization/how_to_use_the_localizers`
+
+.. _registry: http://martinfowler.com/eaaCatalog/registry.html
+
+
+Plug on events (datagrid, savers, remover, etc.)
+""""""""""""""""""""""""""""""""""""""""""""""""
+It is the best extensibility point you have.
+Unfortunately, it is not always possible as Akeneo PIM can't know where you want to plug your code.
+If you can not, it does not matter, you can decorate or extend your class and add the extension point you need.
+By the way, feel free to contribute opening an issue to discuss with our team on our `github repository`_
+or let us know your need on our `forum`_. It is important for us to improve ourselves and our application.
```

Comment:
I don't see the point of "Unfortunately [...] you need." 
Same for "If you can't, it doesn't matter". You should say something like "In case there are no events where you can plug your code, you can decorate [...]